components/ingest-service/pipeline/publisher/config_mgmt_publisher.go

```diff
@@ -0,0 +1,92 @@
+package publisher
+
+import (
+	"context"
+	"time"
+
+	"github.com/chef/automate/api/interservice/cfgmgmt/request"
+	cfgmgmt "github.com/chef/automate/api/interservice/cfgmgmt/service"
+	"github.com/chef/automate/components/ingest-service/pipeline/message"
+	log "github.com/sirupsen/logrus"
+)
+
+func BuildConfigMgmtPublisher(client cfgmgmt.CfgMgmtClient, numPublishers int) message.ChefActionPipe {
+	return func(in <-chan message.ChefAction) <-chan message.ChefAction {
+		if numPublishers <= 0 || client == nil {
+			log.Info("Direct publish to config management service is disabled")
+			return actionsNoop(in)
+		}
+
+		out := make(chan message.ChefAction, 100)
+		log.Infof("Starting config-management-service publisher")
+		for i := 0; i < numPublishers; i++ {
+			go configMgmtPublisher(in, client, out)
+		}
+		return out
+	}
+}
+
+func configMgmtPublisher(in <-chan message.ChefAction, client cfgmgmt.CfgMgmtClient, out chan<- message.ChefAction) {
+	for msg := range in {
+		cmpHandleMessage(client, msg, out)
+
+	}
+	close(out)
+}
+
+func cmpHandleMessage(client cfgmgmt.CfgMgmtClient, msg message.ChefAction, out chan<- message.ChefAction) {
+	if err := msg.Ctx.Err(); err != nil {
+		msg.FinishProcessing(err)
+		return
+	}
+
+	defer message.PropagateChefAction(out, &msg)
+
+	if !isPolicyUpdateAction(msg) {
+		return
+	}
+
+	start := time.Now()
+
+	req := &request.PolicyUpdateAction{
+		PolicyName:         msg.InternalChefAction.EntityName,
+		PolicyGroup:        msg.InternalChefAction.ParentName,
+		PolicyRevisionId:   msg.InternalChefAction.RevisionId,
+		ChefServerFqdn:     msg.InternalChefAction.ServiceHostname,
+		ChefServerOrgname:  msg.InternalChefAction.OrganizationName,
+		ChefServerUsername: msg.InternalChefAction.RequestorName,
+		PolicyfileContent:  msg.InternalChefAction.Data,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	_, err := client.HandlePolicyUpdateAction(ctx, req)
+
+	logCtx := log.WithFields(log.Fields{
+		"message_id":  msg.ID,
+		"buffer_size": len(out),
+		"dur":         time.Since(start),
+	})
+
+	if err != nil {
+		logCtx.WithError(err).Error("Failed to publish actions message to config management service")
+		return
```

Comment:
If it fails to publish the action to the config-mgmt-service we should fail the pipeline with `msg.FinishProcessing(err)`

Reply:
👍 will do, I wasn't sure if we wanted to try a best effort to pass it to the next publisher